### PR TITLE
Added Admin, Developer, ProjectManager, and Tester classes with corresponding unit tests

### PR DESCRIPTION
**Pull request by ivanmartin91**
This pull request adds four classes - Admin, Developer, ProjectManager, and Tester - all inheriting from the User class. 

Basic unit tests are included to validate their behavior and ensure proper functionality.

### CODE REVIEW

File: PMSCore/Admin.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    {
+    }
+
+    /// <summary>
+    /// Basic validation to ensure Admin-specific functionality can be extended later.
+    /// </summary>
+    public bool ValidateAdminAccess()
+    {
+        return UserPermissions.Contains(Permission.CanGenerateUserReport);
+    }
```
Comment (DistroRickDev) [inline PMSCore/Admin.cs]: If you create an IUser you'll never be able to access this method there is already an I base implementation that generates if the user in question has permissions to do so
https://github.com/DistroRickDev/PMS-Core/blob/37be584ce667f3e979435d344a47816fdac9188a/PMSCore/User.cs#L131

File: PMSCore/Developer.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+    }
+
+    /// <summary>
+    /// Ensures Developers can only manage tasks and not projects.
+    /// </summary>
+    public bool ValidateDeveloperScope()
+    {
+        return !UserPermissions.Contains(Permission.CanCreateProject);
+    }
```
Comment (DistroRickDev) [inline PMSCore/Developer.cs]: Same as comment above

File: PMSCore/ProjectManager.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    {
+    }
+
+    /// <summary>
+    /// Verifies Project Managers cannot manage users directly.
+    /// </summary>
+    public bool ValidateUserManagementRestriction()
+    {
+        return !UserPermissions.Contains(Permission.CanChangeUser) &&
+               !UserPermissions.Contains(Permission.CanDeleteUser);
```
Comment (DistroRickDev) [inline PMSCore/ProjectManager.cs]: ProjectManager manages projects not user that is admin work I believe also it's already implmented

File: PMSCore/Tester.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    /// <summary>
+    /// Ensures Testers focus only on task verification and reporting.
+    /// </summary>
+    public bool ValidateTesterScope()
+    {
+        return !UserPermissions.Contains(Permission.CanCreateProject) &&
+               !UserPermissions.Contains(Permission.CanDeleteTask);
+    }
```
Comment (DistroRickDev) [inline PMSCore/Tester.cs]: same as comment above

**Review by DistroRickDev — CHANGES_REQUESTED** [review]
LGTM but the users only need to add permissions on constructor all validations are done in the User abstract class
Needs more testing like DeveloperFailsToCreateUser and so one a few corner cases need to be validated still

**Comment by ivanmartin91** [comment]
hi @DistroRickDev , I have implemented the suggested changes, tnx

**Review by DistroRickDev — APPROVED** [review]
LGTM
